src/jdk.compiler/share/classes/com/sun/tools/javac/code/Symbol.java

```diff
@@ -1504,16 +1504,30 @@ public RecordComponent getRecordComponent(VarSymbol field) {
         }
 
         public RecordComponent getRecordComponent(JCVariableDecl var, boolean addIfMissing, List<JCAnnotation> annotations) {
+            RecordComponent toRemove = null;
             for (RecordComponent rc : recordComponents) {
                 /* it could be that a record erroneously declares two record components with the same name, in that
                  * case we need to use the position to disambiguate
                  */
                 if (rc.name == var.name && var.pos == rc.pos) {
-                    return rc;
+                    if (rc.type.hasTag(TypeTag.ERROR) && !var.sym.type.hasTag(TypeTag.ERROR)) {
+                        // Found a wrong record component: save it so that we can remove it later.
+                        // If the class type of the record component is generated by annotation processor, it should
+                        // use the new actual class type and symbol instead of the old dummy ErrorType.
+                        // toRemove = rc;
```

Comment:
there is dead code here, `toRemove` doesn't seem to be assigned ever, probably you uploaded an intermediate patch?

Reply:
The `// toRemove = rc;` should be `toRemove = rc;`. It is a statement instead of a comment. I revised it mitakenly in my second commit `Polish`. In the second commit `Polish`, I only want to fix the indentation so that I didn't test again after revising. Unfortunately, the IDE changed `toRemove = rc;` to a comment. Apology for the mistake.